authn/src/lib.rs

```diff
@@ -120,8 +121,7 @@ pub enum SomeError {
     FailedToRunOpenssl,
 }
 
-fn generate_root_ca() -> Result<Vec<u8>, SomeError> {
-    let my_dir = env!("PWD");
+fn generate_root_ca(my_dir: String) -> Result<Vec<u8>, SomeError> {
```

Comment:
why not a &str?

Reply:
was getting lifetime issues and i don't quite understand them yet.